Source/Core/Common/Subscribable.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <algorithm>
+#include <cstdint>
+#include <functional>
+
+#include "Common/NonCopyable.h"
+
+template <typename... Args>
+class Subscribable : NonCopyable
```

Comment:
This should likely have basic unit tests as well, since it's an addition to Common.